Core/src/org/sleuthkit/autopsy/datamodel/accounts/Accounts.java

```diff
@@ -201,234 +196,289 @@ protected void addNotify() {
-                super.addNotify();
-                refreshKeys();
-            }
+        @Subscribe
+        @Override
+        void handleReviewStatusChange(ReviewStatusChangeEvent event) {
+            refresh(true);
+        }
 
+        @Subscribe
+        @Override
+        void handleDataAdded(ModuleDataEvent event) {
+            refresh(true);
         }
 
-        public AccountsRootNode() {
-            super(Children.LEAF, Lookups.singleton(Accounts.this));
-            setChildren(Children.createLazy(AccountTypeFactory::new));
-            setName(Accounts.NAME);
-            setDisplayName(Bundle.Accounts_RootNode_displayName());
-            this.setIconBaseWithExtension("org/sleuthkit/autopsy/images/accounts.png");    //NON-NLS
+        @Override
+        protected boolean createKeys(List<String> list) {
+            try (SleuthkitCase.CaseDbQuery executeQuery = skCase.executeQuery(
+                    "SELECT DISTINCT blackboard_attributes.value_text as account_type "
+                    + " FROM blackboard_attributes "
+                    + " WHERE blackboard_attributes.attribute_type_id = " + BlackboardAttribute.ATTRIBUTE_TYPE.TSK_ACCOUNT_TYPE.getTypeID());
+                    ResultSet resultSet = executeQuery.getResultSet()) {
+                while (resultSet.next()) {
+                    String accountType = resultSet.getString("account_type");
+                    list.add(accountType);
+                }
+            } catch (TskCoreException | SQLException ex) {
+                LOGGER.log(Level.SEVERE, "Error querying for account_types", ex);
+            }
+
+            return true;
         }
 
         @Override
-        public boolean isLeafTypeNode() {
-            return false;
+        protected Node[] createNodesForKey(String key) {
+            try {
+                String accountType = key;
+                if (accountType.equals(Account.Type.CREDIT_CARD.getTypeName())) {
+                    return new Node[]{new CreditCardNumberAccountTypeNode()};
+                } else {
+                    String accountTypeDisplayname;
+                    try {
+                        accountTypeDisplayname = skCase.getCommunicationsManager().getAccountType(accountType).getDisplayName();
+                    } catch (TskCoreException ex) {
+                        LOGGER.log(Level.SEVERE, "Error getting display name for account type. ", ex);
+                        accountTypeDisplayname = accountType;
+                    }
+
+                    return new Node[]{new DefaultAccountTypeNode(key, accountTypeDisplayname)};
+                }
+            } catch (IllegalArgumentException ex) {
+                LOGGER.log(Level.WARNING, "Unknown account type: {0}", key);
+                //Flesh out what happens with other account types here.
+                return new Node[]{new DefaultAccountTypeNode(key, key)};
+            }
```

Comment:
What is throwing an unchecked IllegalArgumentException that requires us to do this catch?

Reply:
I'm not really sure what if anything is throwing the IllegalArgumentException. 

@millmanorama: Jonathan do you recall?

Reply:
I am not sure.  It looks like nothing throws it now.  Maybe when Account.Type was an enum and if we were looking it up by name or something we threw an exception if we got an unexpected type.  I *think* we can remove the catch block.

Reply:
That was my thought too Nothing in the current code looked like it could throw it. I took out the catch block and tested it, and of course it worked. But since its a unchecked exception, I wanted to make sure.